low-level/files/opt/seagate/sspl/setup/sspl_test.py

```diff
@@ -38,31 +39,37 @@ def __init__(self, args: list):
         self.plan = "self_primary"
         self.avoid_rmq = False
         self.dbus_service = DbusServiceHandler()
-        # Load global, sspl and test configs
-        Conf.load(SSPL_CONFIG_INDEX, sspl_config_path)
-        Conf.load(SSPL_TEST_CONFIG_INDEX, sspl_test_config_path)
-        # Take copy of global config passed to sspl_test
         if args.config and args.config[0]:
-            sspl_test_gc_url = args.config[0]
+            self.sspl_test_gc_url = args.config[0]
         else:
-            sspl_test_gc_url = global_config_path
-        sspl_test_gc_copy_file = "/etc/sspl_test_gc_url.yaml"
-        with open(sspl_test_gc_copy_file, "w") as f:
-            f.write("")
-        self.sspl_test_gc_copy_url = "yaml://%s" % sspl_test_gc_copy_file
-        Conf.load(SSPL_TEST_GLOBAL_CONFIG, self.sspl_test_gc_copy_url)
-        Conf.load("global_config", sspl_test_gc_url)
-        Conf.copy("global_config", SSPL_TEST_GLOBAL_CONFIG)
+            self.sspl_test_gc_url = global_config_path
+        self.sspl_test_gc_copy_file = "/etc/sspl_test_gc_url.yaml"
 
     def validate(self):
         """Check for required packages are installed."""
+        # RPM dependency
+        rpm_deps = {
+            "cortx-sspl-test": None
+            }
         # python 3rd party package dependency
         pip3_3ps_packages_test = {
             "Flask": "1.1.1"
             }
         pkg_validator = PkgV()
-        pkg_validator.validate_pip3_pkgs(host=None, pkgs=pip3_3ps_packages_test,
-            skip_version_check=False)
+        pkg_validator.validate_pip3_pkgs(host=socket.getfqdn(),
+            pkgs=pip3_3ps_packages_test, skip_version_check=False)
+        pkg_validator.validate_rpm_pkgs(host=socket.getfqdn(),
+            pkgs=rpm_deps, skip_version_check=True)
+        # Load global, sspl and test configs
+        Conf.load(SSPL_CONFIG_INDEX, sspl_config_path)
+        Conf.load(SSPL_TEST_CONFIG_INDEX, sspl_test_config_path)
+        # Take copy of supplied config passed to sspl_test and load it
+        with open(self.sspl_test_gc_copy_file, "w") as f:
+            f.write("")
+        self.sspl_test_gc_copy_url = "yaml://%s" % self.sspl_test_gc_copy_file
+        Conf.load(SSPL_TEST_GLOBAL_CONFIG, self.sspl_test_gc_copy_url)
```

Comment:
whats the point of creating & then loading this blank config file? With 3 node deployment, we were doing away with any global config file & its copy right, why needed now?

Reply:
`we were doing away with any global config file & its copy right, why needed now?` Yes, we have global config copy already. But we can not expect same global_config_copy will still be passed to sspl_test.

`whats the point of creating & then loading this blank config file?`
Blank file, later will have global config copy in it.
`sspl_test --config $url` takes provisioner supplied or test purpose user supplied config one. sspl_test used to overwrite some config for test purpose. To avoid touching provisioner supplied config, we copy it and use it.